ui/lib/pki/addon/routes/configuration/edit.js

```diff
@@ -4,5 +4,30 @@
  */
 
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
+import { withConfirmLeave } from 'core/decorators/confirm-leave';
 
-export default class PkiConfigurationEditRoute extends Route {}
+@withConfirmLeave('model.config', ['model.urls', 'model.crl'])
+export default class PkiConfigurationEditRoute extends Route {
+  @service secretMountPath;
+
+  model() {
+    const { urls, crl, engine } = this.modelFor('configuration');
+    return hash({
+      engineId: engine.id,
+      urls,
+      crl,
+    });
```

Comment:
I'm not sure you would need to use `hash` here. Since the configuration route is the parent the model should already be loaded. Although since you used hash there I wonder if `urls` and `crl` are promises? This is more of a question than anything 😄

Reply:
Nope, those should just be record from the parent model,